File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.py
class Solution:
  def subarraySum(self, nums: List[int], k: int) -> int:
    map = {0 : 1}
    sum = 0
    ans = 0
    for v in nums:
      sum += v
      ans += map.get(sum-k, 0)
      map[sum] = map.get(sum, 0) + 1
    return ans
      
    

File: 560-subarray-sum-equals-k/560-subarray-sum-equals-k.scala
object Solution {
  def subarraySum(nums: Array[Int], k: Int): Int = {
    val map = collection.mutable.HashMap[Int, Int](0 -> 1)
    var ans = 0
    var sum = 0
    for (v <- nums) {
      sum += v
      if (map.contains(sum-k)) {
        ans += map(sum-k)
      }
      if (map.contains(sum))
        map.put(sum, map(sum)+1)
      else
        map.put(sum, 1)
    }
    ans
  }
}